robocop/checkers/naming.py

```diff
@@ -720,6 +747,10 @@ def check_setting_name(self, name, node):
                 "setting-name-not-in-title-case", setting_name=name, node=node, col=col + 1, end_col=col + len(name) + 1
             )
 
+    def check_task_related_settings(self, name, node):
+        if "test" in name.lower() and not "task" in name.lower():
```

Comment:
We also dont support translations here for RF 6+

Reply:
Also looking at condition - can name contain both task and test in name? Do we need to check for both?

Reply:
Oh, you're right. I was thinking to first filter out all settings like `Documentation`, which don't have test- or task-related variants. But, yeah, no sense in doing that.

Reply:
Fixed.